config/initializers/sidekiq.rb

```diff
@@ -6,14 +6,25 @@ def self.redis_url
 
 Sidekiq.configure_client do |config|
   config.redis = { url: SidekiqConfig.redis_url }
+  config.client_middleware do |chain|
+    chain.add SidekiqUniqueJobs::Middleware::Client
+  end
 end
 
 Sidekiq.configure_server do |config|
   config.redis = { url: SidekiqConfig.redis_url }
+
+  config.client_middleware do |chain|
+    chain.add SidekiqUniqueJobs::Middleware::Client
+  end
+
   config.server_middleware do |chain|
     chain.add Sidekiq::Congestion::Limiter
+    chain.add SidekiqUniqueJobs::Middleware::Server
```

Comment:
I think this should probably go ahead of the `Sidekiq::Congestion::Limiter` as that particular middleware only runs server side and relies on redis stored sets for limiting, https://github.com/parrish/sidekiq-congestion. As such the values in the redis sets will eventually time out etc even with no other jobs runs etc as they use redis key TTL values, see https://github.com/parrish/Congestion/pull/1

This means that if the client middleware of UniqueJobs runs and say the `Sidekiq::Congestion::Limiter` middleware stops the server callback running it may leave the lock state in an unexpected state. 

Rather i'd put preference on the `SidekiqUniqueJobs::Middleware::Client` to run first / always and then allow the `Sidekiq::Congestion::Limiter` to run 

Details on middleware ordering - https://github.com/mperham/sidekiq/wiki/Middleware#middleware-ordering

Happy to expand on this on a call or here as well.

Reply:
ah this was something i did question (the ordering that is) when i was configuring the middleware for SidekiqUniqueJobs. 

thanks for context.